tasks.py

```diff
@@ -0,0 +1,217 @@
+from contextlib import contextmanager
+import os
+from shutil import rmtree, copytree, copy as copyfile
+from tempfile import mkdtemp
+
+from invoke import task
+from invoke.exceptions import Failure
+
+
+supported_combinations = {
+    '2.7': {'pyside', 'pyqt', 'wx', 'null'},
+    '3.5': {'pyqt', 'null'},
+}
+
+dependencies = {
+    "numpy",
+    "pandas",
+    "pygments",
+    "traits",
+    "nose",
+    "coverage",
+}
+
+extra_dependencies = {
+    'pyside': {'pyside'},
+    'pyqt': {'pyqt'},
+    'wx': {'wxpython'},
+    'null': set()
+}
+
+environment_vars = {
+    'pyside': {'ETS_TOOLKIT': 'qt4', 'QT_API': 'pyside'},
+    'pyqt': {'ETS_TOOLKIT': 'qt4', 'QT_API': 'pyqt'},
+    'wx': {'ETS_TOOLKIT': 'wx'},
+    'null': {'ETS_TOOLKIT': 'null'},
+}
+
+
+@task
+def install(ctx, runtime='3.5', toolkit='null', environment=None, edm='edm'):
+    """ Install traitsui and dependencies into a clean EDM environment. """
+    if toolkit not in supported_combinations.get(runtime, set()):
+        msg = "Python {} and toolkit {} not supported by test environments"
+        raise RuntimeError(msg.format(runtime, toolkit))
+
+    if environment is None:
+        environment = 'traitsui-test-{}-{}'.format(runtime, toolkit)
+
+    packages = dependencies | extra_dependencies.get(toolkit, set())
+
+    print("Creating environment '{}'".format(environment))
+    ctx.run("{} install -y -e '{}' --version '{}' {}".format(
+        edm, environment, runtime, ' '.join(packages)))
+
+    # use current master of pyface
+    clone_into_env(ctx, environment, 'pyface', edm=edm)
+
+    # install traitsui from this directory
+    run_in_env(ctx, environment, "python setup.py install", edm=edm)
+
+
+@task
+def test(ctx, runtime='3.5', toolkit='null', environment=None, edm='edm'):
+    """ Run the test suite in a given environment with the specified toolkit """
+    if toolkit not in supported_combinations.get(runtime, set()):
+        msg = "Python {} and toolkit {} not supported by test environments"
+        raise RuntimeError(msg.format(runtime, toolkit))
+
+    if environment is None:
+        environment = 'traitsui-test-{}-{}'.format(runtime, toolkit)
+
+    try:
+        # simple test to see if designated environment exists
+        ctx.run("{} prefix -e '{}'".format(edm, environment))
+    except Failure:
+        # environment doesn't exist, so create it and populate
+        print("Environment not found, creating...")
+        install(ctx, runtime, toolkit, environment, edm)
+
+    envvars = environment_vars.get(toolkit, {}).copy()
+
+    # run tests
+    with do_in_tempdir(files=['.coveragerc'], capture_files=['.coverage']):
+        run_in_env(ctx, environment,
+                   "coverage run -m nose.core -v traitsui.tests",
```

Comment:
Does coverage support the `u` option for unbuffered output? If not then I would suggest to set the related python environment variable.